### DISCUSSION TITLE
AzuraCast, Autocue and master_me (or StereoTool, or other LADSPA plugins)

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#7179] [Show and tell]
## AzuraCast, Autocue and _master_me_ (or others)

AzuraCast can use LADSPA plugins like [_master_me_](https://github.com/trummerschlunk/master_me) (and others), and also Thimeo’s [_StereoTool_](https://www.thimeo.com/stereo-tool/).

I’ve run several tests with _master_me_ on Rolling Release #[5bbd484](https://github.com/AzuraCast/AzuraCast/commit/5bbd484fe31128df8de8040212f0c3a22683fb81) (2024-06-06 6:21), because there seemed to be some uncertainty about the "right" settings when using Autocue as well.

### In short

- Works nicely here, adding an extra load average of ~0.5–1 per station (expected). YMMV.
- Use Autocue’s default **-18 LUFS** target as input.
- Set _master_me_’s Loudness Target to your desired playout LUFS target.

#### Why Autocue -18 LUFS?

- It’s the default, for a reason.
- If you let Autocue _write tags_ to files, they will be saved with the correct loudness target of -18 LUFS, which is the generally approved standard used by nearly all software. It also corresponds to the—now obsolete—old ReplayGain 1/mp3gain "89 dB".
- It’s a perfectly good _input loudness_ for post-processing.

#### _Master_me_ settings

- _Master_me_ (and Stereo Tool, for that matter) come _last_ in the playout chain, so _they_ define the final output loudness.
- Set your desired playout target _there_.
  - EBU recommends -23 LUFS. _Do_ use that for video streams, but in radio, almost no one uses it.
  - I’m a known [**-18 LUFS Radio**](https://github.com/AzuraCast/AzuraCast/discussions/7177) advocate, and EBU has -18 listed as "temporarily allowed". So I will always use that value.
  - _Spotify_ use -14 LUFS, you could also use that.
- For my tests, I used both **-18** and **-14** LUFS, each for several hours on a real station.
- I used the predefined _Music General_ setting, which has these parameters:
  ```
  radio = ladspa.master_me(
      bypass = false,
      target = -14,
      brickwall_bypass = false,
      brickwall_ceiling = -1.00,
      brickwall_release = 75.00,
      eq_bypass = false,
      eq_highpass_freq = 5.00,
      eq_side_bandwidth = 1.00,
      eq_side_freq = 600.00,
      eq_side_gain = 1.00,
      eq_tilt_gain = 0.00,
      gate_attack = 0.00,
      gate_bypass = true,
      gate_hold = 50.00,
      gate_release = 430.50,
      gate_threshold = -90.00,
      kneecomp_attack = 20.00,
      kneecomp_bypass = false,
      kneecomp_dry_wet = 50,
      kneecomp_ff_fb = 50,
      kneecomp_knee = 6.00,
      kneecomp_link = 60,
      kneecomp_makeup = 0.00,
      kneecomp_release = 340.00,
      kneecomp_strength = 20,
      kneecomp_tar_thresh = -4.00,
      leveler_brake_threshold = -10.00,
      leveler_bypass = false,
      leveler_max = 10.00,
      leveler_max__ = 10.00,
      leveler_speed = 20,
      limiter_attack = 3.00,
      limiter_bypass = false,
      limiter_ff_fb = 50,
      limiter_knee = 3.00,
      limiter_makeup = 0.00,
      limiter_release = 40.00,
      limiter_strength = 80,
      limiter_tar_thresh = 6.00,
      mscomp_bypass = false,
      high_attack = 8.00,
      high_crossover = 8000.00,
      high_knee = 12.00,
      high_link = 30,
      high_release = 30.00,
      high_strength = 30,
      high_tar_thresh = -12.00,
      low_attack = 15.00,
      low_crossover = 60.00,
      low_knee = 12.00,
      low_link = 70,
      low_release = 150.00,
      low_strength = 10,
      low_tar_thresh = -3.00,
      makeup = 1.00,
      dc_blocker = false,
      input_gain = 0.00,
      mono = false,
      phase_l = false,
      phase_r = false,
      stereo_correct = false,
      radio
  )
  ```
- Feel free to use your own setup, but **_please know what you’re doing!_** It’s much easier making your sound real _awful_ than making it _beautiful_.

### It’s advanced. It needs _computing power_.

- Be aware that _both_ Autocue and tools like _master_me_ do highly advanced computations which _need_ raw CPU power!
- _Stereo Tool_ especially is known to be a CPU hog. No wonder, considering the functionality it has.
- Especially on station startup, this will hit your machine _hard_, since Liquidsoap has to ensure at least _one track from each playlist_ has to be readily available for _immediate_ playout in case of fallbacks. Thus, it will autocue that many files when starting up! Just imagine, I have ~70 playlists per station and 4 stations, so it’ll cue up 4x70=280 tracks at startup, making my load average go up to ~9 on a 4-CPU system for about 5–7 minutes. It then goes down again to the usual ~0.7–1, or 1–1.5 with _master_me_ enabled.
- Personally, I’m okay with that, since the benefits clearly outweigh having to wait a little for station startup. Both AzuraCast and Liquidsoap are known to be real stable and can run for a long time (470+ days, my best) without needing a restart, so no problem here.
- If you have _large tracks_ like multi-hour pre-recorded shows or DJ sets, this could make your station inoperable if you don’t have enough raw CPU power available, and fast access to files.

### Conclusion

- Yes, you can use Autocue and a post-processing tool just fine with AzuraCast.
- Just know the pitfalls and set up everything appropriately.

**Comment by mattrackley23-ai** [comment]
Hi - Could i ask a question please ? I am using the cloud version of Azuracast for my online radio station and I have Autocue enabled along with Audio Post Processing set to Master_Me for Music General. My LUFS is set to -9. Personally theses work well for my set up, but from reading all the posts on here, its made me wonder if i am doing it all correctly. I have also uploaded a Stereotools preset as well but im questioning if that is actually being used. Any thoughts and or comments on this set up would be greatly received. Thanks Matt
